### ISSUE TITLE
VM only: Script functions stored in variables are treated as bound methods when using other

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#7913]
### Description
This is related to the changes made in #7349. In VM only, storing a script function in a variable will cause `other` to behave as if it was a method. This happens even if you specifically use `method` with undefined as the first argument to unbind the function from any scope. If the script functions are not stored in a variable and instead called directly, it works as expected and `other` resolves like it does it YYC.

### Steps To Reproduce
```
//in a script
function func(){
	show_debug_message($"In func: {object_get_name(other.object_index)}");
	nested_func_in_variable();
}

function nested_func() {
	show_debug_message($"In nested_func: {object_get_name(other.object_index)}");
}
```

```
//In Object1
with Object2 {
	show_debug_message($"In with: {object_get_name(other.object_index)}");
	func_in_variable();
}
```
```
//In Object2
func_in_variable = method(undefined, func);
nested_func_in_variable = method(undefined, nested_func);
```
In YYC, it prints the following:
```
In with: Object1
In func: Object1
In nested_func: Object1
```
In VM, it instead prints this:
```
In with: Object1
In func: Object2
In nested_func: Object2
```
If you call the functions directly instead of storing them in variables first, then VM prints the same thing as YYC.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.634 Runtime v2024.1100.0.658

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
Windows

[3a336d39-0a52-4f1f-8527-8044e807f4ac](https://api.gamemaker.io/api/github/downloads/3a336d39-0a52-4f1f-8527-8044e807f4ac)

**Comment by rwkay** [comment]
Fixed in 2024.11

**Comment by Emc1923** [comment]
Verified in  IDE v2024.1100.0.639  Runtime v2024.1100.0.662.
